File: java-spring-virtual-threads/benchmark/src/main/java/com/example/benchmark/model/Circle.java
package com.example.benchmark.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Circle implements FigureInterface{
    private final int diameter;

    @JsonCreator
    Circle(@JsonProperty("diameter") int diameter) {
        this.diameter = diameter;
    }

    @Override
    public String getType() {
        return "circle";
    }

    @Override
    public double computeArea() {
        var r = (double)diameter / 2;
        return Math.PI * Math.pow(r, 2);
    }
}

File: kotlin-ktor-jvm/src/main/kotlin/com/example/model/Circle.kt
package com.example.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.math.pow

@Serializable
@SerialName("circle")
final class Circle(
    private val diameter: Int
): Figure {
    override fun getType(): String {
        return "circle";
    }

    override fun computeArea(): Double {
        val r = diameter.toDouble() / 2
        return (Math.PI * r.pow(2.0))
    }
}